core/nodejsActionBase/runner.js

```diff
@@ -79,12 +81,31 @@ class NodeActionRunner {
     run(args) {
         return new Promise((resolve, reject) => {
             try {
-                var result = this.userScriptMain(args);
+
+                (async () => {
+                    // This will choose one idle worker in the pool
+                    // to execute your heavy task without blocking
+                    // the main thread!
+                    console.log("Running action in a thread!")
+                    
+                    const res = await pool.exec({
+                        task: this.userScriptMain,
+                        workerData: {
```

Comment:
we should set `param: args` to the arguments are passed to the actual function.

Reply:
isn't it `params` which should be passed to the action function?

Reply:
Oh, I've got what you meant. Actually API changed even further. Updated